librz/util/buf.c

```diff
@@ -408,37 +408,44 @@ RZ_API bool rz_buf_append_buf_slice(RzBuffer *b, RzBuffer *a, ut64 offset, ut64
 	return res;
 }
 
-// return an heap-allocated string read from the RzBuffer b at address addr. The
-// length depends on the first '\0' found in the buffer. If there is no '\0' in
-// the buffer, there is no string, thus NULL is returned.
-RZ_API char *rz_buf_get_string(RzBuffer *b, ut64 addr) {
-	const int MIN_RES_SZ = 64;
-	ut8 *res = RZ_NEWS(ut8, MIN_RES_SZ + 1);
-	ut64 sz = 0;
-	st64 r = rz_buf_read_at(b, addr, res, MIN_RES_SZ);
-	bool null_found = false;
-	while (r > 0) {
-		const ut8 *needle = rz_mem_mem(res + sz, r, (ut8 *)"\x00", 1);
-		if (needle) {
-			null_found = true;
-			break;
+#define GET_STRING_BUFFER_SIZE 32
+
+RZ_API char *rz_buf_get_nstring(RzBuffer *b, ut64 addr, size_t size) {
+	RzStrBuf *buf = rz_strbuf_new(NULL);
+
+	while (true) {
+		char tmp[GET_STRING_BUFFER_SIZE];
+		st64 r = rz_buf_read_at(b, addr, (ut8 *)tmp, RZ_ARRAY_SIZE(tmp));
```

Comment:
```suggestion
		st64 r = rz_buf_read_at(b, addr, (ut8 *)tmp, sizeof(tmp));
```

Reply:
Also, here there should be a condition to choose between `sizeof(tmp)` and the remaining bytes to read. Otherwise I think you could end up with a longer string than what you requested.

Reply:
Why should I use sizeof(tmp) and not RZ_ARRAY_SIZE?

Reply:
> Also, here there should be a condition to choose between `sizeof(tmp)` and the remaining bytes to read. Otherwise I think you could end up with a longer string than what you requested.

It doesn't seems to break anything, the last test is testing the corner case.

Reply:
RZ_ARRAY_SIZE gets the number of elements in an array... with chars it is the same as sizeof, but what you really want in this case is imho the all number of bytes. It seems conceptually better (to me at least).